### ISSUE TITLE
Upgrade Multus to v4.x

### ISSUE THREAD

**Issue by chomatdam** [pelotech/foundation#90]
## Context
v4 is doing a split between thin (as v3) and thick (adding metrics)

Migrate to thick and resolve any issues encountered

## Notes for the migration:
_Notes gathered when trying to upgrade to 4.0.2, some items may have been resolved since_

- They moved from a [python container image](https://github.com/k8snetworkplumbingwg/multus-cni/blob/v3.9.3/images/Dockerfile) on 3.x to a [distroless image](https://github.com/k8snetworkplumbingwg/multus-cni/blob/v4.0.2/images/Dockerfile) on 4.x (`cat` not available anymore)
- Their [thick container image](https://github.com/k8snetworkplumbingwg/multus-cni/blob/v4.0.2/images/Dockerfile.thick) uses a slim Debian where we would still have `cat`  for the startupProbe
- Add `-thick` to the overridden container tag to not have their too loose versioning
- `--multus-conf-file` not available on their thick version, being a flag only for [their thin entrypoint](https://github.com/k8snetworkplumbingwg/multus-cni/blob/v4.0.2/cmd/thin_entrypoint/main.go)

**Comment by scoquelin** [comment]
Was able to confirm some of those points locally : 

```
# kubectl exec -it multus-multus-cni-hxfcc -n kube-system -- /bin/sh
Defaulted container "multus-cni" out of: multus-cni, install-multus-binary (init), generate-kubeconfig (init)
# /opt/bitnami/multus-cni/bin/multus --version
multus: version:v4.1.2(clean,released), commit:adfb270991d410417a71755d01f53445510ef2a0, date:2024-10-08T02:15:30+00:00
# whereis cat
cat: /usr/bin/cat
```

**Comment by scoquelin** [comment]
Here is some documentation attempt on how to configure `multusMasterCNI` with the thick version : https://github.com/k8snetworkplumbingwg/multus-cni/pull/1245
